src/fabric.c

```diff
@@ -913,14 +931,12 @@ static int ofi_layering_ok(const struct fi_provider *provider,
 
 	if ((count == 1) && ofi_is_util_prov(provider) &&
 	    !ofi_has_util_prefix(prov_vec[0])) {
-		if (!strcasecmp(prov_vec[0], "sockets")) {
+		core_ofi_prov = ofi_getprov(prov_vec[0], strlen(prov_vec[0]));
+		if (core_ofi_prov &&
+		    ofi_disable_util_layering(core_ofi_prov->provider)) {
```

Comment:
I *think* the issue with the crashes could be coming from here.  There is a list of known providers using struct ofi_prov.  That list exists prior to providers actually be registered with the core.  I think the check above needs to be:
```
if (core_ofi_prov && core_ofi_prov->provider && ofi_disable_util_layering(...))
```

Reply:
Yep, that sounds right after looking at the stack trace:

```
(gdb) bt
#0  0x00007ffb7bcbd6fc in ofi_disable_util_layering (provider=0x0) at src/fabric.c:145
#1  0x00007ffb7bcbf8c6 in ofi_layering_ok (provider=0x7ffb7c08ba00 <rxm_prov>, prov_vec=0x55989a90d720, count=1, flags=0) at src/fabric.c:936
#2  0x00007ffb7bcbfc54 in fi_getinfo_ (version=65548, node=0x0, service=0x0, flags=0, hints=0x55989a900c50, info=0x7ffcd41dd6c8) at src/fabric.c:996
#3  0x000055989876a226 in run (hints=0x55989a900c50, node=0x0, port=0x0) at util/info.c:314
#4  0x000055989876a660 in main (argc=3, argv=0x7ffcd41dd818) at util/info.c:429
(gdb) p *core_ofi_prov
$3 = {next = 0x55989a9012d0, prov_name = 0x55989a9012b0 "udp", provider = 0x0, dlhandle = 0x0, hidden = false}
```

Reply:
I think there's a case sensitivity issue here. The UDP provider is initialized by the time this check happens. Changing `udpx_prov->name` from 'UDP' to 'udp' fixes the crash. I'm not sure if this check should ignore case or if there's a bug earlier in the `getinfo()` path.

Reply:
The ofi_getprov() call should use strncasecmp instead of strncmp.  As it is, we'll end up with 2 entries, one for UDP and the other for udp.  I think that issue is separate, but exposes the missing check that's needed here.  That is, if you tried using something like the psm provider on your system, where it wouldn't be available, you should hit the same problem.

For good measure, I would rename the UDP provider to udp to be consistent with all other names being lower case.

Can you make these changes as part of this series?

Reply:
Sure, I've made the changes. Let me know if I missed anything.